### ISSUE TITLE
have install.py use venv in python3

### ISSUE THREAD

**Issue by nerdvegas** [nerdvegas/rez#982]
Currently the installation uses a bundled virtualenv lib in build_utils/ to create the venv-based rez installation. This is failing in py 3.9+.

Instead, detect py3 and switch to its native venv instead.

**Comment by nerdvegas** [comment]
Related: https://github.com/nerdvegas/rez/issues/980

Also, unrelated installation bug reported separately:

```
% uname -a
Darwin LHR-AL-22482 19.6.0 Darwin Kernel Version 19.6.0: Thu Oct 29 22:56:45 PDT 2020; root:xnu-6153.141.2.2~1/RELEASE_X86_64 x86_64
% which python
/usr/local/bin/python # installed via homebrew
% python --version
Python 3.9.0
% cd rez-2.69.6
% python ./install.py -v /usr/local/rez
installing rez to /usr/local/rez...
ERROR: The executable /private/tmp/bin/python3.9 is not functioning
ERROR: It thinks sys.prefix is '/usr/local/Cellar/python@3.9/3.9.0_1/Frameworks/Python.framework/Versions/3.9' (should be '/usr/local/rez')
ERROR: virtualenv is not compatible with this system or executable
```

**Comment by JeanChristopheMorinPerso** [comment]
So I was initially saying in Slack that we shouldn't use the builtin `venv`, but I retract that. Between Python 3.3 and 3.6, I wouldn't have recommended to use the builtin one, but with 3.6+, I'd say it's safe to use.

See https://virtualenv.pypa.io/en/latest/.

It would require us some changes, but I think it's doable. @nerdvegas I can give it a shot tomorrow night.

**Comment by nerdvegas** [comment]
See https://github.com/nerdvegas/rez/pull/1006, seems to work.

Note that I've only switched to venv on py 3.7+. I ran into issues with 3.6. For some reason, venv installs pip v18.2, and this causes rez-pip to fail (I don't recall why rez-pip needs pip>=19, but I know there's good reason). I tried doing a pip install --upgrade in the venv but hit some pretty odd problems. I could've kept digging, but it seemed more straightforward to just leave things as they are for 3.6, since it's already working.
